tests/ui/traits/new-solver/negative-coherence-bounds.rs

```diff
@@ -0,0 +1,27 @@
+// check-pass
+
+#![feature(specialization)]
```

Comment:
what is that testing/why does this fail rn. please add a comment 

r=me afterwards